renderer.php

```diff
@@ -141,6 +141,14 @@ public function formulation_and_controls(question_attempt $qa, question_display_
                     array('class' => 'validationerror'));
         }
 
+        // ISS1053 - Output current language
+        // to be returned by validaion request and also stored in
+        // question step data on submission.
+        $fieldname = $qa->get_qt_field_name('step_lang');
+        $currentlang = current_language();
+        $result .= html_writer::empty_tag('input', array('type' => 'hidden',
+                    'name' => $fieldname, 'value' => $currentlang));
+
```

Comment:
Yes, this ties to the UI language, not to the question language, which might be different, i.e., it's good that you did not mix `stack_multilang` into this.. I wonder, though, if this is always the same as the `<html>`-tags lang`-attribute. STACK-JS already syncs that over to the iframe, and if this would ever be different, then it would need to sync this instead.

Reply:
Iframe block is doing
`$code .= '<html xmlns="http://www.w3.org/TR/xhtml1/strict" lang="' . 
            stack_get_system_language() . '">';`
which is essentially the same thing. Is that what you're referring to or is there something else I'm missing?

Reply:
Nothing, I am hallucinating something; I had a memory that I actually read that value from the VLE DOM. But then that would have been slow and pointless.